src/utils/default-drives-manager.ts

```diff
@@ -0,0 +1,215 @@
+import {
+    BaseDocumentDriveServer,
+    DefaultRemoteDriveInfo,
+    DocumentDriveServerOptions,
+    RemoveOldRemoteDrivesOption
+} from '../server';
+import { DriveNotFoundError } from '../server/error';
+import { requestPublicDrive } from './graphql';
+
+export class DefaultDrivesManager {
+    private defaultRemoteDrives = new Map<string, DefaultRemoteDriveInfo>();
+    private removeOldRemoteDrivesConfig: RemoveOldRemoteDrivesOption;
+
+    constructor(
+        private server: BaseDocumentDriveServer,
+        options?: Pick<
+            DocumentDriveServerOptions,
+            'defaultRemoteDrives' | 'removeOldRemoteDrives'
+        >
+    ) {
+        if (options?.defaultRemoteDrives) {
+            for (const defaultDrive of options.defaultRemoteDrives) {
+                this.defaultRemoteDrives.set(defaultDrive.url, {
+                    ...defaultDrive,
+                    status: 'PENDING'
+                });
+            }
+        }
+
+        this.removeOldRemoteDrivesConfig = options?.removeOldRemoteDrives || {
```

Comment:
Maybe we could already add support for env variables at the document-drive level?
In Connect we would set `DOCUMENT_DRIVE_OLD_REMOTE_DRIVES_STRATEGY` and so on.